### PR TITLE
[device/marvell] Mitigation for security vulnerability

### PR DESCRIPTION
**Pull request by maipbui**
Signed-off-by: maipbui <maibui@microsoft.com>

Dependency: [PR (#12065)](https://github.com/sonic-net/sonic-buildimage/pull/12065) needs to merge first.

#### Why I did it
`os` and `commands` modules are not secure against maliciously constructed input
`getstatusoutput` is detected without a static string, uses `shell=True`
#### How I did it
Eliminate the use of `os` and `commands`
Use `subprocess` instead
#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [X] 201911
- [X] 202006
- [X] 202012
- [X] 202106
- [X] 202111
- [X] 202205

### CODE REVIEW

File: device/marvell/armhf-marvell_et6448m_52x-r0/plugins/sfputil.py [hunk 1]
```diff
@@ -113,8 +109,8 @@ def get_presence(self, port_num):
         pos = [1, 2, 4, 8]
         bit_pos = pos[prt]
         if smbus_present == 0:
-            cmdstatus, sfpstatus = commands.getstatusoutput(
-                'i2cget -y 0 0x41 0x3')  # need to verify the cpld register logic
+            p = subprocess.run(['i2cget', '-y', '0', '0x41', '0x3'], capture_output=True, universal_newlines=True)  # need to verify the cpld register logic
```
Comment (qiluo-msft) [inline device/marvell/armhf-marvell_et6448m_52x-r0/plugins/sfputil.py:112]: >p [](http://example.com/codeflow?start=12&length=1)

LGTM actually catches a true bug in your code. You overwrite `p` and this is a behavior change.

File: device/marvell/arm64-marvell_db98cx8580_32cd-r0/plugins/sfputil.py [hunk 2]
```diff
@@ -31,9 +26,10 @@ class SfpUtil(SfpUtilBase):
     _qsfp_ports = list(range(_port_start, ports_in_block + 1))
 
     def __init__(self):
-        os.system("modprobe i2c-dev")
+        subprocess.run(["modprobe", "i2c-dev"])
         if not os.path.exists("/sys/bus/i2c/devices/0-0050"):
-            os.system("echo optoe2 0x50 > /sys/bus/i2c/devices/i2c-0/new_device")
+            with open("/sys/bus/i2c/devices/i2c-0/new_device", 'w') as f:
+                subprocess.run(["echo", "optoe2", "0x50"], stdout=f)
```
Comment (qiluo-msft) [inline device/marvell/arm64-marvell_db98cx8580_32cd-r0/plugins/sfputil.py:32]: >"echo", "optoe2", "0x50" [](http://example.com/codeflow?start=32&length=24)

I see the original intention is to write a string to a file, so there is no need to invoke subprocess at all.

File: device/marvell/armhf-marvell_et6448m_52x-r0/plugins/sfputil.py (hunk 1 above)
Comment (qiluo-msft) [reply device/marvell/armhf-marvell_et6448m_52x-r0/plugins/sfputil.py:112]: Suggest use a long name for one-shot usage in internal block. The less used, the longer the name.

File: device/marvell/x86_64-marvell_db98cx8580_16cd-r0/plugins/sfputil.py [hunk 3]
```diff
@@ -31,9 +26,10 @@ class SfpUtil(SfpUtilBase):
     _qsfp_ports = list(range(_port_start, ports_in_block + 1))
 
     def __init__(self):
-        os.system("modprobe i2c-dev")
+        subprocess.run(["modprobe", "i2c-dev"])
         if not os.path.exists("/sys/bus/i2c/devices/0-0050"):
-            os.system("echo optoe2 0x50 > /sys/bus/i2c/devices/i2c-0/new_device")
+            with open("/sys/bus/i2c/devices/i2c-0/new_device", 'w') as f:
+                subprocess.run(["echo", "optoe2", "0x50"], stdout=f)
```
Comment (qiluo-msft) [inline device/marvell/x86_64-marvell_db98cx8580_16cd-r0/plugins/sfputil.py:32]: >"echo", "optoe2", "0x50" [](http://example.com/codeflow?start=32&length=24)

Same: write string to file.

File: device/marvell/x86_64-marvell_db98cx8580_32cd-r0/plugins/sfputil.py [hunk 4]
```diff
@@ -31,9 +26,10 @@ class SfpUtil(SfpUtilBase):
     _qsfp_ports = list(range(_port_start, ports_in_block + 1))
 
     def __init__(self):
-        os.system("modprobe i2c-dev")
+        subprocess.run(["modprobe", "i2c-dev"])
         if not os.path.exists("/sys/bus/i2c/devices/0-0050"):
-            os.system("echo optoe2 0x50 > /sys/bus/i2c/devices/i2c-0/new_device")
+            with open("/sys/bus/i2c/devices/i2c-0/new_device", 'w') as f:
+                subprocess.run(["echo", "optoe2", "0x50"], stdout=f)
```
Comment (qiluo-msft) [inline device/marvell/x86_64-marvell_db98cx8580_32cd-r0/plugins/sfputil.py:32]: >subprocess.run(["echo", "optoe2", "0x50"], stdout=f) [](http://example.com/codeflow?start=16&length=52)

The same.

File: device/marvell/armhf-marvell_et6448m_52x-r0/plugins/sfputil.py (hunk 1 above)
Comment (maipbui) [reply device/marvell/armhf-marvell_et6448m_52x-r0/plugins/sfputil.py]: Right... I changed the variable name.

File: device/marvell/arm64-marvell_db98cx8580_32cd-r0/plugins/sfputil.py (hunk 2 above)
Comment (maipbui) [reply device/marvell/arm64-marvell_db98cx8580_32cd-r0/plugins/sfputil.py]: All fixed, please check again.

File: device/marvell/x86_64-marvell_db98cx8580_16cd-r0/plugins/sfputil.py (hunk 3 above)
Comment (qiluo-msft) [reply device/marvell/x86_64-marvell_db98cx8580_16cd-r0/plugins/sfputil.py]: Use `with`

File: device/marvell/arm64-marvell_db98cx8580_32cd-r0/plugins/sfputil.py (hunk 2 above)
Comment (maipbui) [reply device/marvell/arm64-marvell_db98cx8580_32cd-r0/plugins/sfputil.py]: Done.

**Comment by qiluo-msft** [comment]
@Sabareesh-Kumar-Anandan @antony-rheneus @shilimkarvg Could you help verify and review this PR?

**Review by qiluo-msft — APPROVED** [review]

File: platform/marvell-armhf/sonic-platform-nokia/7215/sonic_platform/component.py [hunk 5]
```diff
@@ -144,7 +130,15 @@ def get_firmware_version(self):
             return self._get_cpld_version(self.index)
 
         if self.index == 1:
-            cmdstatus, uboot_version = cmd.getstatusoutput('grep --null-data U-Boot /dev/mtd0ro|head -1 | cut -d" " -f2-4')
+            cmd1 = ['grep', '--null-data', 'U-Boot', '/dev/mtd0ro']
+            cmd2 = ['head', '-1']
+            cmd3 = ['cut', '-d', ' ', '-f2-4']
+            with subprocess.Popen(cmd1, universal_newlines=True, stdout=subprocess.PIPE) as p1:
```
Comment (qiluo-msft) [inline platform/marvell-armhf/sonic-platform-nokia/7215/sonic_platform/component.py:136]: >with subprocess.Popen(cmd1, universal_newlines=True, stdout=subprocess.PIPE) as p1: [](http://example.com/codeflow?start=12&length=83)

Possible to reuse 3 cmds function?

**Comment by qiluo-msft** [comment]
>     import os

Nokia's folder should be in a separate PR.

---
Refers to: platform/marvell-armhf/sonic-platform-nokia/7215/sonic_platform/chassis.py:9 in e2aef69. [](commit_id = e2aef69dfb204f9ed368e5e7cd5443e432b6e870, deletion_comment = False)

**Review by qiluo-msft — APPROVED** [review]

**Review by qiluo-msft — APPROVED** [review]

**Comment by maipbui** [comment]
@antony-rheneus Could you help review and verify?

**Pull request merged**